transform_encoder.py

```diff
@@ -0,0 +1,131 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Dict, Generator, Optional, Tuple
+
+import numpy as np
+import torch
+from jina import DocumentArray, Executor, requests
+from jina.logging.predefined import default_logger as logger
+from transformers import AutoModel, AutoTokenizer
+
+
+class TransformerTorchEncoder(Executor):
+    """
+    The transformer torch encoder encodes sentences into embeddings.
+    """
+
+    def __init__(
+        self,
+        pretrained_model_name_or_path: str = "sentence-transformers/distilbert-base-nli-stsb-mean-tokens",
+        base_tokenizer_model: Optional[str] = None,
+        pooling_strategy: str = "mean",
+        layer_index: int = -1,
+        max_length: Optional[int] = None,
+        embedding_fn_name: str = "__call__",
+        device: str = "cpu",
+        default_traversal_path: str = "r",
+        default_batch_size: int = 32,
+        *args,
+        **kwargs,
+    ):
+        """
+        :param pretrained_model_name_or_path:
+        :param base_tokenizer_model:
+        :param pooling_strategy:
+        :param layer_index:
+        :param max_length:
+        :param embedding_fn_name:
+        :param device:
+        :param default_traversal_path: Used in the encode method an define traversal on the received `DocumentArray`
+        :param default_batch_size: Defines the batch size for inference on the loaded PyTorch model.
+        :param args:
+        :param kwargs:
+        """
+        super().__init__(*args, **kwargs)
+        self.default_traversal_path = default_traversal_path
+        self.default_batch_size = default_batch_size
+        self.pretrained_model_name_or_path = pretrained_model_name_or_path
+        self.base_tokenizer_model = (
+            base_tokenizer_model or pretrained_model_name_or_path
+        )
+        self.pooling_strategy = pooling_strategy
+        self.layer_index = layer_index
+        self.max_length = max_length
+        if not device in ["cpu", "cuda"]:
+            logger.error("Torch device not supported. Must be cpu or cuda!")
+            raise RuntimeError("Torch device not supported. Must be cpu or cuda!")
+        if device == "cuda" and not torch.cuda.is_available():
+            logger.warning(
```

Comment:
instantiate a logger. It should not be logged by the `default` one. It is not a core issue

Reply:
https://github.com/jina-ai/executor-transformer-torch-encoder/pull/1/commits/9116b7c45458db6495de5eca7b8f2d31dd4104c1